packages/react-components/src/components/RichTextEditor/RichTextEditor.tsx

```diff
@@ -423,9 +423,29 @@ export const RichTextEditor = React.forwardRef<RichTextEditorComponentRef, RichT
     /* @conditional-compile-remove(rich-text-editor-image-upload) */
     setPreviousInlineImages(prevInlineImage);
 
+    const colorTransformFunction = (
+      lightColor: string
+      /* baseLValue?: number,
+      colorType?: 'text' | 'background' | 'border',
+      element?: HTMLElement */
+    ): string => {
+      if (!isDarkThemed(theme)) {
+        return lightColor;
+      }
+      // For dark mode, switch black to be white
+      switch (lightColor) {
+        case '--darkColor_rgb_0_0_0_':
+          return theme.palette.neutralPrimary;
+
+        default:
+          return lightColor;
+      }
+    };
+
     const initialModel = createEditorInitialModel(initialContent, contentModel);
     if (editorDiv.current) {
       editor.current = new Editor(editorDiv.current, {
+        generateColorKey: colorTransformFunction,
         inDarkMode: isDarkThemed(theme),
         // doNotAdjustEditorColor is used to disable default color and background color for Rooster component
```

Comment:
I'm wondering if the issue exists because the `doNotAdjustEditorColor` doesn't do anything anymore https://github.com/microsoft/roosterjs/blob/76750e24f5733f64519e82d85cd3cf76bbd73dc3/packages/roosterjs-content-model-core/lib/corePlugin/lifecycle/LifecyclePlugin.ts#L43